runtime/vm/options.c

```diff
@@ -0,0 +1,240 @@
+	} else { \
+		o->howset = (defaultSource); \
+	}
+
+/*
+ * More usable SET_OPTION for simple options that have no command line aliases
+ * See parameter definitions for SET_OPTION0
+ */
+#define SET_OPTION(option, cmdLineOption, vtype, setValue, defaultSource) \
+	SET_OPTION0(option, cmdLineOption, vtype, setValue, defaultSource, )
+
+/*
+ * Version of SET_OPTION when two command line options alias to one another
+ * The presence of either command line option will initialize the option with HOWSET_COMMANDLINE
+ * See parameter definitions for SET_OPTION0
+ * Also: aliasCmdLineOption is a string for a command line option that is an alias for cmdLineOption
+ */
+#define SET_OPTION_WITH_ALIAS(option, cmdLineOption, aliasCmdLineOption, vtype, setValue, defaultSource) \
+	SET_OPTION0(option, cmdLineOption, vtype, setValue, defaultSource, \
+		|| (findArgInVMArgs( PORTLIB, vm->vmArgsArray, STARTSWITH_MATCH, aliasCmdLineOption, NULL, 0) >= 0))
+
+/*
+ * Following two macros can be used for options whose default values are HOWSET_ERGONOMIC and HOWSET_DEFAULT
+ * See parameter definitions for SET_OPTION0
+ */
+
+#define SET_ERGONOMIC_OPTION(option, cmdLineOption, vtype, value) SET_OPTION(option, cmdLineOption, vtype, value, SET_ERGONOMIC)
+#define SET_DEFAULT_OPTION(option, cmdLineOption, value) SET_OPTION(option, cmdLineOption, value, SET_DEFAULT)
+
+/*
+ * Following two macros can be used for -X options whose default values are HOWSET_ERGONONIC and HOWSET_DEFAULT
+ * See parameter definitions for SET_OPTION0
+ * cmdLineOption is derived automatically from the name provided for option
+ */
+#define SET_ERGONOMIC_XOPTION(option, vtype, value) SET_OPTION(IDX_VMOPT_ ## option, VMOPT_ ## X ## option, vtype, value, SET_ERGONOMIC)
+#define SET_DEFAULT_XOPTION(option, vtype, value) SET_OPTION(IDX_VMOPT_ ## option, VMOPT_ ## X ## option, vtype, value, SET_DEFAULT)
+
+/*
+ * Following two macros can be used for -XX options whose default values are HOWSET_ERGONONIC and HOWSET_DEFAULT
+ * See parameter definitions for SET_OPTION0
+ * cmdLineOption is derived automatically from the name provided for option
+ */
+#define SET_ERGONOMIC_XXOPTION(option, vtype, value) SET_OPTION(IDX_VMOPT_ ## option, VMOPT_ ## XX ## option, vtype, value, SET_ERGONOMIC)
+#define SET_DEFAULT_XXOPTION(option, vtype, value) SET_OPTION(IDX_VMOPT_ ## option, VMOPT_ ## XX ## option, vtype, value, SET_DEFAULT)
+
+void initializeFlags(J9JavaVM* vm) {
+	J9OptionDesc *o;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	SET_OPTION_WITH_ALIAS(IDX_VMOPT_MAXHEAPSIZE, VMOPT_XMX, VMOPT_XXMAXHEAPSIZE, v_sizet, vm->memoryManagerFunctions->j9gc_get_maximum_heap_size(vm), SET_ERGONOMIC);
+	SET_ERGONOMIC_XXOPTION(MAXDIRECTMEMORYSIZE, v_uint64, vm->directByteBufferMemoryMax);
+}
+
+/* Prints final flags (options) to the console. The list of options printed is not yet complete. */
+void printFlags(J9JavaVM* vm) {
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+        j9tty_printf(PORTLIB, "[Global flags]\n");
```

Comment:
This should use a NLS string. Similarly the option_category_string[] and option_howset_string[].
Also, pls fix the indenting here.

Reply:
damn, another indent problem....sorry I troubled you with those and thanks for catching it.

On NLS: I had imagined this was more like diagnostic output in the same category as, say, verbose logging where I don't believe we employ NLS strings. I'll build the NLS strings if you think that reasoning is faulty or inconsistent with other similar kinds of options?

Reply:
We could go either way on this one. If the output is meant to be machine parsable rather than human readable, not using NLS is better.

Reply:
Looking at -verbose:sizes, the code is using NLS strings.

Reply:
certainly elasticsearch is parsing the output, but whether NLS would interfere with that, I haven't checked.